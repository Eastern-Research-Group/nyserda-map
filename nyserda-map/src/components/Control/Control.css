.nyserda-control {
    position: absolute;
    left: 12px;
    top: 7rem;
    padding: 1rem;
    width: 12rem;
    /*height: 20vh;*/
    z-index: 400;
    background-color: white;
    border: 1px solid black;
    box-shadow: 5px 5px 2.5px #888888;
}

.nyserda-control .nyserda-data-selector {
    margin: 1rem 0;
}

.nyserda-control .nyserda-data-selector select {
    width: 100%;
    height: 2rem;
    font-size: 1.2rem;
}

.nyserda-control .nyserda-layer-selector {
    /*margin: 1rem 0;*/
    margin: 4rem 0 1rem;
}

.nyserda-slider-caption {
    position: relative;
    display: block;
    top: 2.5rem;
    margin: 0 auto;
    text-align: center;
}

.nyserda-control .nyserda-layer-checkbox {
    display: block;
    padding: .25rem 0;
}

.nyserda-type-checkboxes {
    margin-top: 4rem;
}

.nyserda-type-checkbox {
    top: 0;
    left: 0;
    padding: .5rem 0;
    margin: 0;
    text-align: left;
    border: none;
}

.nyserda-type-checkbox input[type="checkbox"] {
    display: inline-block;
    vertical-align: top;
}

.nyserda-checkbox-text {
    display: inline-block;
    width: 90%;
}

.rangeslider__label {
    margin-left: -.5rem;
    margin-top: 1.2rem;
    -webkit-transform: rotate(270deg);
    -moz-transform: rotate(270deg);
    -o-transform: rotate(270deg);
    -ms-transform: rotate(270deg);
    transform: rotate(270deg);
}

.rangeslider-horizontal {
    border-radius: 0;
}

.rangeslider-horizontal {
    height: 5px;
}

.rangeslider-horizontal .rangeslider__fill {
    height: 5px;
}

.rangeslider-horizontal .rangeslider__handle {
    /*margin-left: 12px;*/
    margin-left: 6px;
    width: 5px !important;
    height: 25px !important;
    border-radius: 0 !important;
}

.rangeslider-horizontal .rangeslider__handle::after {
    display: none;
}
