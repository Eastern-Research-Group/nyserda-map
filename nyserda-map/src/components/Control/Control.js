import React from 'react';
import Slider from 'react-rangeslider';
import 'react-rangeslider/lib/index.css';
import {layerTypes} from '../../redux/modules/control';
import './Control.css';


export default props => {
    const labels = {};
    props.years.forEach(year => labels[year] = year);
    return (
        <div className="nyserda-control">
            <div className="nyserda-data-selector">
                <select
                    name="data-selector"
                    id="data-selector"
                    value={props.area}
                    onChange={props.onSelectArea}
                >
                    <option value="" hidden>Select an area</option>
                    <option value="nassau">Nassau</option>
                    <option value="nyc">New York City</option>
                    <option value="westchester">Westchester</option>
                </select>
            </div>

            <div className={`nyserda-slider`}>
                <Slider
                    min={props.years[0]}
                    max={props.years[props.years.length - 1]}
                    value={props.selectedYear}
                    onChange={props.onChangeSlider}
                    labels={labels}
                    orientation="horizontal"/>
                {props.years.length !== 0 ? (
                    <span className="nyserda-slider-caption">
                        Year
                    </span>
                ) : null}

            </div>

            {props.years.length !== 0 ? (
                <div className="nyserda-type-checkboxes">
                    <fieldset className="nyserda-type-checkbox">
                        <label htmlFor="existing-marsh">
                            <input
                                type="checkbox"
                                id="existing-marsh"
                                value={layerTypes.EXISTING_MARSH}
                                checked={props.selectedLayerTypes.indexOf(layerTypes.EXISTING_MARSH) !== - 1}
                                onChange={props.onSelectLayerType}/>
                            <div className="nyserda-checkbox-text">
                                Marshland that existed in 2008
                            </div>
                        </label>
                    </fieldset>

                    <fieldset className="nyserda-type-checkbox">
                        <label htmlFor="new-coastal-marsh">
                            <input
                                type="checkbox"
                                id="new-coastal-marsh"
                                value={layerTypes.NEW_COASTAL_MARSH}
                                checked={props.selectedLayerTypes.indexOf(layerTypes.NEW_COASTAL_MARSH) !== - 1}
                                onChange={props.onSelectLayerType}/>
                            <div className="nyserda-checkbox-text">
                                Marshland that could emerge due to sea-level rise
                            </div>
                        </label>
                    </fieldset>
                </div>
            ) : null}


            {/*<div className="nyserda-layer-selector" id="nyserda-layers-container">*/}
            {/*{props.categories.map((category, key) => (*/}
            {/*<label*/}
            {/*key={key}*/}
            {/*htmlFor={`category-${key}`}*/}
            {/*className='nyserda-layer-checkbox'>*/}
            {/*<input*/}
            {/*type="checkbox"*/}
            {/*id={`category-${key}`}*/}
            {/*value={category}*/}
            {/*checked={props.selectedCategories.indexOf(category) !== - 1}*/}
            {/*onChange={props.onSelectCategory}/>*/}

            {/*{category}*/}

            {/*</label>*/}
            {/*))}*/}
            {/*</div>*/}
        </div>
    );
}
