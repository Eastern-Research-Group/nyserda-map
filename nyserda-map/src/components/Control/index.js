import {connect} from 'react-redux';
import Control from './Control';
import {
    getArea,
    getCategories,
    getSelectedCategories,
    getYears,
    getSelectedYear,
    getSelectedLayerTypes,
} from '../../redux/modules/reducer';
import {
    selectArea,
    selectCategory,
    selectYear,
    selectLayerType,
} from '../../redux/modules/control';

const mapStateToProps = state => ({
    area: getArea(state),
    categories: getCategories(state),
    selectedCategories: getSelectedCategories(state),
    years: getYears(state),
    selectedYear: getSelectedYear(state),
    selectedLayerTypes: getSelectedLayerTypes(state),
});

const mapDispatchToProps = dispatch => ({
    onSelectArea(e) {
        dispatch(selectArea(e.target.value));
    },
    onSelectCategory(e) {
        const {value, checked} = e.target;
        dispatch(selectCategory(value, checked));
    },
    onChangeSlider(value) {
        dispatch(selectYear(value));
    },
    onSelectLayerType(e) {
        const {value, checked} = e.target;
        dispatch(selectLayerType(value, checked));
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Control)