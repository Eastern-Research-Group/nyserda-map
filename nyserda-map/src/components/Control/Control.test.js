import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Control from './Control';

function setup () {
    const props = {
        example: 'FooBarBaz',
        onChangeExample: jest.fn(),
    };

    const enzymeWrapper = shallow(<Control {...props}/>);

    return {
        props,
        enzymeWrapper,
    };
}

describe('<Control />', () => {
    it('renders without crashing', () => {
        const {props} = setup();
        const component = renderer.create(<Control {...props}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});