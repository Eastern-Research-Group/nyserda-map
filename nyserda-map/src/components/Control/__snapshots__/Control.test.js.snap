// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Control /> renders without crashing 1`] = `
<div
  className="nyserda-control"
>
  <div
    className="nyserda-data-selector"
  >
    <select
      id="data-selector"
      name="data-selector"
    >
      <option
        hidden={true}
        value=""
      >
        Select an area
      </option>
      <option
        value="nassau"
      >
        Nassau
      </option>
      <option
        value="nyc"
      >
        New York City
      </option>
      <option
        value="westchester"
      >
        Westchester
      </option>
    </select>
  </div>
  <div
    className="nyserda-layer-selector"
    id="nyserda-layers-container"
  />
</div>
`;
